source/core/rtc_adapter/VideoReceiveAdapter.cc

```diff
@@ -0,0 +1,296 @@
+            video_recv_config.decoders.clear();
+            /*
+            if (!video_send_config.rtp.rtx.ssrcs.empty()) {
+              video_recv_config.rtp.rtx_ssrc = video_send_config.rtp.rtx.ssrcs[i];
+              video_recv_config.rtp.rtx_associated_payload_types[kSendRtxPayloadType] =
+                  video_send_config.rtp.payload_type;
+            }
+            */
+            webrtc::VideoReceiveStream::Decoder decoder;
+            decoder.decoder_factory = this;
+            // Add VP8 decoder
+            decoder.payload_type = VP8_90000_PT;
+            decoder.video_format = webrtc::SdpVideoFormat(
+                webrtc::CodecTypeToPayloadString(webrtc::VideoCodecType::kVideoCodecVP8));
+            RTC_DLOG(LS_INFO) <<  "Config add decoder:" << decoder.ToString();
+            video_recv_config.decoders.push_back(decoder);
+            // Add VP9 decoder
+            decoder.payload_type = VP9_90000_PT;
+            decoder.video_format = webrtc::SdpVideoFormat(
+                webrtc::CodecTypeToPayloadString(webrtc::VideoCodecType::kVideoCodecVP9));
+            RTC_DLOG(LS_INFO) <<  "Config add decoder:" << decoder.ToString();
+            video_recv_config.decoders.push_back(decoder);
+            // Add H264 decoder
+            decoder.payload_type = H264_90000_PT;
+            decoder.video_format = webrtc::SdpVideoFormat(
+                webrtc::CodecTypeToPayloadString(webrtc::VideoCodecType::kVideoCodecH264));
+            RTC_DLOG(LS_INFO) <<  "Config add decoder:" << decoder.ToString();
+            video_recv_config.decoders.push_back(decoder);
+            // Add H265 decoder
+            decoder.payload_type = H265_90000_PT;
+            decoder.video_format = webrtc::SdpVideoFormat(
+                webrtc::CodecTypeToPayloadString(webrtc::VideoCodecType::kVideoCodecH265));
+            RTC_DLOG(LS_INFO) <<  "Config add decoder:" << decoder.ToString();
+            video_recv_config.decoders.push_back(decoder);
+
+            RTC_DLOG(LS_INFO) << "VideoReceiveStream::Config " << video_recv_config.ToString();
+            m_videoRecvStream = call()->CreateVideoReceiveStream(std::move(video_recv_config));
+            m_videoRecvStream->Start();
+            call()->SignalChannelNetworkState(webrtc::MediaType::VIDEO, webrtc::NetworkState::kNetworkUp);
+        }
+    });
+}
+
+void VideoReceiveAdapterImpl::requestKeyFrame() {
+    // m_videoRecvStream->GenerateKeyFrame();
+    m_isWaitingKeyFrame = true;
+}
+
+std::vector<webrtc::SdpVideoFormat> VideoReceiveAdapterImpl::GetSupportedFormats() const {
+    return std::vector<webrtc::SdpVideoFormat>{
+        webrtc::SdpVideoFormat(
+            webrtc::CodecTypeToPayloadString(webrtc::VideoCodecType::kVideoCodecVP8)),
+        webrtc::SdpVideoFormat(
+            webrtc::CodecTypeToPayloadString(webrtc::VideoCodecType::kVideoCodecVP9)),
+        webrtc::SdpVideoFormat(
```

Comment:
SdpVideoFormat should include all profiles it supported instead of single H264 SdpVideoFromat.

Reply:
No params matches all profiles here